app/services/cache/mapping/declaration/DeclarationBuilder.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.cache.mapping.declaration
+
+import forms.declaration.RepresentativeDetails
+import services.cache.ExportsCacheModel
+import services.mapping.declaration._
+import wco.datamodel.wco.dec_dms._2.Declaration
+import wco.datamodel.wco.declaration_ds.dms._2.{DeclarationFunctionalReferenceIDType, DeclarationTypeCodeType}
+
+object DeclarationBuilder {
+
+  val defaultFunctionCode = "9"
+
+  def build(exportsCacheModel: ExportsCacheModel): Declaration = {
+    val declaration = new Declaration()
+
+    declaration.setFunctionCode(FunctionCodeBuilder.build(defaultFunctionCode))
+    declaration.setFunctionalReferenceID(createDeclarationFunctionalReferenceIDType(exportsCacheModel))
+
+    val typeCode: Option[DeclarationTypeCodeType] = exportsCacheModel.additionalDeclarationType.map(
+      additionalDeclarationType =>
+        TypeCodeBuilder.createTypeCode(additionalDeclarationType, exportsCacheModel.dispatchLocation)
+    )
+
+    declaration.setTypeCode(typeCode.getOrElse(new DeclarationTypeCodeType()))
+    declaration.setGoodsItemQuantity(GoodsItemQuantityBuilder.createGoodsItemQuantity(exportsCacheModel.items.toSeq))
+
+    generateAgent(exportsCacheModel).foreach { data =>
+      declaration.setAgent(data)
+    }
+
+    //    declaration.setGoodsShipment(GoodsShipmentBuilder.build)
```

Comment:
We should not commit commented code to master, even if we are going to replace them.
Will this cause the smoke tests to fail as well?